src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogJenkinsPublisher.java

```diff
@@ -122,4 +133,16 @@ private PluginData collectPluginData(Jenkins instance) {
 
         return pluginData.build();
     }
+
+    @SuppressRestrictedWarnings({ UpdateSiteWarningsMonitor.class, UpdateSite.Warning.class })
+    private Integer getPluginWarningsCount() {
+        try {
+            UpdateSiteWarningsMonitor updateSiteWarningsMonitor = ExtensionList.lookupSingleton(UpdateSiteWarningsMonitor.class);
+            Map<PluginWrapper, List<UpdateSite.Warning>> activePluginWarningsByPlugin = updateSiteWarningsMonitor.getActivePluginWarningsByPlugin();
+            return activePluginWarningsByPlugin.size();
```

Comment:
Do we know if this is thread safe?

Reply:
I checked the `getActivePluginWarningsByPlugin()` method's implementation, and the map that it returns is local, so there's no other thread that can reach it.